libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -356,6 +383,15 @@ bool AP_BattMonitor::overpower_detected(uint8_t instance) const
 #endif
 }
 
+bool AP_BattMonitor::has_cell_voltages(const uint8_t instance) const
+{
+    if (instance < _num_instances && drivers[instance] != nullptr) {
+        return drivers[instance]->has_cell_voltages();
```

Comment:
The comment from @WickedShell is actually in the wrong place. This method is fine, the one in SMBus backend might not be - I think this is just the equivalent of `has_current` which doesn't say that the current has been read, but that it supports reading current. So I think it is OK that the SMBus backend doesn't check if the cell voltages have actually been read.

One change I think should be done (for `has_current` and `has_cell_voltages`) is to put the methods in each of the SMBus drivers, because logically makes more sense each driver saying what it supports - the base SMBus class provides common methods for reading stuff but it doesn't actually read anything, the drivers do.